fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/mixin/networking/ServerLoginNetworkHandlerMixin.java

```diff
@@ -79,6 +96,16 @@ private int removeLateCompressionPacketSending(MinecraftServer server) {
 		return -1;
 	}
 
+	@Inject(method = "onCookieResponse", at = @At("HEAD"))
+	private void triggerCookieFuture(CookieResponseC2SPacket packet, CallbackInfo ci, @Share("allowedCookie") LocalBooleanRef allowedCookie) {
```

Comment:
Why exactly is this shared over just being computed in the WrapWithCondition?

Reply:
Yeah Idk what I was doing, this has been cleaned up